### ISSUE TITLE
'pm install' does not install packages

### ISSUE THREAD

**Issue by deefrawley** [Flow-Launcher/Flow.Launcher#256]
**Describe the bug/issue**
The new package manager doesn't install packages. I can choose the package, I get the mesasage box with the details, and then the two toaster popups saying the package has been downloaded but Flow doesn't restart and the packages are not available or located in the %APPDATA%/Roaming/FlowLauncher directory. The log files logs nothing about the attempt.

Note I have completely uninstalled Flow and deleted all %APPDATA% directories afterwards for a clean install and issue persists. Running explicitly as Administrator doesn't help (Flow crashed the first time I did this and start typing 'pm' but I haven't been able to reproduced that).

**To Reproduce**
1. Activate Flow
2. Type 'pm install' and choose any package
3. Click 'yes' on the message box showing the plugin details and confirming you want to install
4. Toaster popup saying 'downloading...''
5. Toaster popup saying downloading has finished.
6. Nothing else happens. Enter Flow Settings - Plugins tab and nothing new listed.

**Your System**
```
Windows build number: 10.0.19042.6851
Flow Launcher version: 1.6 
```
**Flow Launcher Error Log**
[2021-01-01.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/5758428/2021-01-01.txt)

**Comment by taooceros** [comment]
@jjw24 
![image](https://user-images.githubusercontent.com/45326534/103431818-46264400-4c11-11eb-976a-f46f8433a985.png)
Seems the translation in catch is false.

**Comment by taooceros** [comment]
> (Flow crashed the first time I did this and start typing 'pm' but I haven't been able to reproduced that)

That may because you access the downloaded manifest before it has downloaded successfully. Will add a default List for access, since plugins are initialized async.

**Comment by taooceros** [comment]
Haven't been able to reproduce the error...... Is it possible for me to provide you with an edited version of plugin manager with more error handling so that you can log the error so we can look up?

**Comment by taooceros** [comment]
[Flow.Launcher.Plugin.PluginsManager.zip](https://github.com/Flow-Launcher/Flow.Launcher/files/5758536/Flow.Launcher.Plugin.PluginsManager.zip)

Take a try on replacing the Flow.Launcher.Plugin.PluginsManager.dll with this one and check whether Logger can record the error?

**Comment by deefrawley** [comment]
Replaced the file and Flow won't even launch now, no log created at all. 
I'm CitizenDee on the Discord server if you wanted to do some other trials/debugging with me over there in DMs

**Comment by taooceros** [comment]
ok, thank you!

**Comment by taooceros** [comment]
[Release.zip](https://github.com/Flow-Launcher/Flow.Launcher/files/5758603/Release.zip)

**Comment by deefrawley** [comment]
Using @taooceros debug DLL I got the following:
```
2021-01-01 15:34:02.3793|ERROR|PluginManager|-------------------------- Begin exception --------------------------
2021-01-01 15:34:02.3793|ERROR|PluginManager|InstallException
2021-01-01 15:34:02.3793|ERROR|PluginManager|Exception full name:
 <System.IO.FileNotFoundException>
2021-01-01 15:34:02.3793|ERROR|PluginManager|Exception message:
 <Could not load file or assembly 'ICSharpCode.SharpZipLib, Version=1.2.0.246, Culture=neutral, PublicKeyToken=1b03e6acf1164f73'. The system cannot find the file specified.>
2021-01-01 15:34:02.4346|ERROR|PluginManager|Exception stack trace:
 <   at Flow.Launcher.Plugin.PluginsManager.Utilities.UnZip(String zipFilePath, String strDirectory, Boolean overwrite)
   at Flow.Launcher.Plugin.PluginsManager.PluginsManager.Install(UserPlugin plugin, String downloadedFilePath)
   at Flow.Launcher.Plugin.PluginsManager.PluginsManager.InstallOrUpdate(UserPlugin plugin)>
2021-01-01 15:34:02.4346|ERROR|PluginManager|Exception source:
 <Flow.Launcher.Plugin.PluginsManager>
2021-01-01 15:34:02.4346|ERROR|PluginManager|Exception target site:
 <Void UnZip(System.String, System.String, Boolean)>
2021-01-01 15:34:02.4346|ERROR|PluginManager|Exception HResult:
 <-2147024894>
2021-01-01 15:34:02.4346|ERROR|PluginManager|-------------------------- End exception --------------------------
```
I have ICSharpCode.SharpZipLib.dll in \AppData\Local\FlowLauncher\app-1.6.0\ so I copied this to \AppData\Local\FlowLauncher\app-1.6.0\Plugins\Flow.Launcher.Plugin.PluginsManager, restarted Flow, and the plugin manager worked and I could install plugins with the pm command.

**Comment by jjw24** [comment]
<Could not load file or assembly 'ICSharpCode.SharpZipLib, Version=1.2.0.246, Culture=neutral, PublicKeyToken=1b03e6acf1164f73' this may be the culprit

so looks like error is when Unzipping the downloaded plugin

**Comment by jjw24** [comment]
> @jjw24
> ![image](https://user-images.githubusercontent.com/45326534/103431818-46264400-4c11-11eb-976a-f46f8433a985.png)
> Seems the translation in catch is download success.

how did this happen lol

**Comment by jjw24** [comment]
Thanks @deefrawley for the info, i am able to replicate the error. @taooceros it works for us because we have the dll in .nuget\packages folder in our profile folder where VS keeps all the references. So if the program is unable to find the dll in the folder,  then it will go to the default path which is in the .nuget folder.

**Comment by jjw24** [comment]
i have a suspicion this is caused by the post build script deleting duplicate dlls. I have put the 1.6.0 release as prerelease which will stop people from updating.

**Comment by jjw24** [comment]
@taooceros i dont understand though, why the error is not logged even if it outputs the wrong msg

**Comment by taooceros** [comment]
> @taooceros i dont understand though, why the error is not logged even if it outputs the wrong msg

It can caused by that Task stop error from throwing, since all Plugins query job is wrapped in a Task. But I am not totally sure

**Comment by jjw24** [comment]
nah it's because the install method is not in the try catch, the method contains the unzip call, which errors out, the throw must be swallowed up further up the chain

**Comment by taooceros** [comment]
> nah it's because the install method is not in the try catch, the method contains the unzip call, which errors out, the throw must be swallowed up further up the chain

That's possible, the one I provided to @deefrawley add the try catch so that the error is logged. I believe the simply throw out error won't be logged.

**Comment by jjw24** [comment]
Just published a rebuild of the 1.6.0 release, should fix the issue where nothing happens when you download plugin.

Will put through a fix for the error message and packaging of sharpzip dependency

**Comment by deefrawley** [comment]
Thanks Jeremy, appreciate the quick response from yourself and taooceros.
